PathsActivity.py

```diff
@@ -84,7 +84,17 @@ def __init__(self, handle):
         self.show_all()
 
         self._game = Game(canvas, parent=self, colors=self.colors)
-        self._setup_presence_service()
+        #self._setup_presence_service()
+        self.connect('shared', self._shared_cb)
+        self.connect('joined', self._joined_cb)
```

Comment:
You can remove this line as you connected the joined signal in collabwrapper to the same callback.

Reply:
Signals mean different things, even though they are both called "joined".  It is best to use the signals and API at the same level of the stack.

Reply:
> @quozl could you clarify what you meant by "It's best to use the signals and API at the same level of the stack" as they're both different signals and both get emitted differently.

CollabWrapper is a software layer, or stack level, above the Telepathy API exposed by the sugar3 classes.  So whenever someone mixes signals and API between Telepathy API and CollabWrapper, there's an opportunity to bypass a layer, which makes code non-modular and confusing.

> My statement above was because as `CollabWrapper` is used for collaboration, we should just call the `joined` event of collabwrapper and not the one from the `sugar3.activity.activity.Activity` class.

Yes, that's sort of the same thing, except I gave reasons?  :grin: